### PR TITLE
Reuse Test Data Across Multiple Tests For Geneformer

### PR DESCRIPTION
**Pull request by nvmvle**
### Description
Refactored test data loading in the bionemo-geneformer test suite to eliminate redundant data downloads and improve test efficiency. Previously, multiple test files were independently loading the same datasets using `load()` calls, leading to repeated downloads and slower test execution. This PR centralizes commonly used test data into session-scoped fixtures in `conftest.py`, ensuring each dataset is loaded only once per test session and shared across all tests that need it.

### Type of changes

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [x]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
# Before: Each test file loaded data independently
from bionemo.testing import load
data_path = load("single_cell/testdata-20241203") / "cellxgene_2023-12-15_small_processed_scdl"

# After: Tests use centralized fixtures
def test_something(data_path):  # data_path fixture automatically injected
    # Use data_path directly - no manual loading required
    assert data_path.exists()
```

### Pre-submit Checklist

 - [x] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [x] I have added/updated tests as needed
 - [x] All existing tests pass successfully

## Summary by CodeRabbit

- Tests
  - Migrated tests to session-scoped fixtures for datasets and checkpoints, reducing redundant loading.
  - Converted tests to fixture-based inputs (replacing hardcoded paths) and updated signatures accordingly.
  - Enhanced training tests with TensorBoard and TFLOPS validation; added CLI test ensuring event logs and checkpoints.
  - Parameterized Stop-and-Go tests for precision/sequence length; added consistency and consumed-samples checks.

- Refactor
  - Standardized data loading via fixtures and updated imports to align with current APIs.
  - Decoupled test helpers from local filesystem assumptions.

- Chores
  - Added a utility to verify TensorBoard logs programmatically.

### CODE REVIEW

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/conftest.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import pytest
 
+from bionemo.core.data.load import load as core_load
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/conftest.py]: please use load() method https://github.com/NVIDIA/bionemo-framework/blob/656c70acb584bb0366faab61fa3a2218e6ed26e7/sub-packages/bionemo-core/src/bionemo/core/data/load.py#L142

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/scripts/test_train_geneformer.py [hunk 2]
```diff
@@ -56,50 +55,91 @@ def test_main_runs(tmpdir, create_checkpoint_callback: bool, data_path: Path):
             result_dir=result_dir,
             wandb_project=None,
             wandb_offline=True,
-            num_steps=5,
-            limit_val_batches=1,
-            val_check_interval=2,
+            num_steps=20,  # Run for more steps to ensure metrics are logged
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/scripts/test_train_geneformer.py]: why to make the training longer?

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/conftest.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
...
 from bionemo.testing.data.load import load
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/conftest.py]: I am not sure what it is, probably load method from core. You can remove load from testing

https://github.com/NVIDIA/bionemo-framework/blob/656c70acb584bb0366faab61fa3a2218e6ed26e7/sub-packages/bionemo-testing/src/bionemo/testing/data/load.py#L17

**Review by dorotat-nv — CHANGES_REQUESTED** [review]

**Comment by nvmvle** [comment]
/ok to test https://github.com/NVIDIA/bionemo-framework/pull/955/commits/85affe87163139ae6902bc6a3979b62892850c41

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 4]
```diff
@@ -894,7 +901,13 @@ def _train_model_get_ckpt(
 
 @pytest.mark.needs_gpu
 def test_continue_from_checkpoint_geneformer(
-    tmpdir, geneformer_config: GeneformerConfig, n_layers_test: int = 3, n_steps_train: int = 100, batch_size: int = 16
+    tmpdir,
+    geneformer_config: GeneformerConfig,
+    data_path,
+    geneformer_nemo1_checkpoint,
+    n_layers_test: int = 3,
+    n_steps_train: int = 100,
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: can we do 5 instead o 100 and batch size ie 1?

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 5]
```diff
@@ -959,7 +976,13 @@ def test_continue_from_checkpoint_geneformer(
 
 @pytest.mark.needs_gpu
 def test_finetune_geneformer(
-    tmpdir, geneformer_config: GeneformerConfig, n_layers_test: int = 3, n_steps_train: int = 100, batch_size: int = 16
+    tmpdir,
+    geneformer_config: GeneformerConfig,
+    data_path,
+    geneformer_nemo1_checkpoint,
+    n_layers_test: int = 3,
+    n_steps_train: int = 100,
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: can we do 5 instead of 100 and batch size 1?

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 6]
```diff
@@ -1020,7 +1047,13 @@ def test_finetune_geneformer(
 @pytest.mark.needs_gpu
 @pytest.mark.skip(reason="PEFT currently broken with fusions activated.")
 def test_finetune_geneformer_with_peft(
-    tmpdir, geneformer_config: GeneformerConfig, n_layers_test: int = 3, n_steps_train: int = 100, batch_size: int = 16
+    tmpdir,
+    geneformer_config: GeneformerConfig,
+    data_path,
+    geneformer_nemo1_checkpoint,
+    n_layers_test: int = 3,
+    n_steps_train: int = 100,
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: can we do steps 5 instead of 100 and batch size 1?

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 7]
```diff
@@ -894,7 +901,13 @@ def _train_model_get_ckpt(
 
 @pytest.mark.needs_gpu
 def test_continue_from_checkpoint_geneformer(
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py:904]: this unit test tested if we can traing for ie 5 steps, finish the training and then continue training from the checkpoint created after 5 steps ie for another 5 steps. hence we do not need geneformer_nemo1_checkpoint as it is not finetuning task but a continuation of training

**Review by dorotat-nv — CHANGES_REQUESTED** [review]

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 8]
```diff
@@ -940,26 +960,34 @@ def test_continue_from_checkpoint_geneformer(
             config=update_base_geneformer_config,  # same config as before since we are just continuing training
             n_steps_train=n_steps_train,
             batch_size=batch_size,
+            data_path=data_path,
+            checkpoint_path=ckpt_path,  # Continue from the checkpoint created in initial training
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: in fact, you dont need `checkpoint_path` here, you should set `root_dir=tmpdir / "pretrain"` only as in the initial training. Simply, the setup you are testing is

1. I run training with command A for 100 steps and stop
2. I want to run training with command A for 10 more steps so I execute commmand A for 110 steps (100 of them were already completed). Please use 5 and 5 steps

Comment (dorotat-nv) [reply sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: With what is done now, the finetuning is tested, which is tested in the next test

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_stop_and_go.py [hunk 9]
```diff
@@ -164,3 +215,9 @@ def test_train_val_init_consumed_samples(self):
         assert val_consumed_go == 0
         assert train_consumed_stop == 0
         assert train_consumed_go > 0
+
+
+def test_save_restore_via_experiment_manager(tmpdir, data_path):
+    """Test save and restore functionality via experiment manager."""
+    # TODO: Implement this test
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_stop_and_go.py]: is this part unfinished?

Comment (nvmvle) [reply sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_stop_and_go.py]: removed redundant function

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py (hunk 8 above)
Comment (nvmvle) [reply sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: updated

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py [hunk 10]
```diff
@@ -914,52 +927,58 @@ def test_continue_from_checkpoint_geneformer(
...
         continue_checkpoint, continue_metrics, continue_trainer = _train_model_get_ckpt(
-            name="test_experiment_continue",
-            root_dir=tmpdir / "continue_training",  # new checkpoint will land in a subdir of this
-            config=update_base_geneformer_config,  # same config as before since we are just continuing training
-            n_steps_train=n_steps_train,
+            name="test_experiment", # same experiment name to trigger resume
+            root_dir=tmpdir / "pretrain",  # new checkpoint will land in a subdir of this
+            config=base_geneformer_config,  # same config as before since we are just continuing training
+            n_steps_train=n_steps_train
+            + 10,  # This simulates running the same training command again but for more total steps
```
Comment (dorotat-nv) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_model.py]: +10 is in the next line, might cause problems
